src/_guides/language/effective-dart/usage_migrated.md

```diff
@@ -1152,6 +1157,48 @@ named parameter whose name doesn't match the name of the field you are
 initializing. But when you *can* use initializing formals, you *should*.
 
 
+### DON'T use `late` when a constructor initialization list will do.
+
+Sound null safety requires Dart to ensure that a non-nullable field must be
+initialized before it can be read. Since fields can be read inside the
+constructor body, this means you get an error if you don't initialize a
+non-nullable field before the body runs.
+
+You can make this error go away by marking the field `late`. That turns the
+compile-time error into a *runtime* error if you access the field before it is
+initialized. That's what you need in some cases, but often the right fix is to
+initialize the field in the constructor initialization list:
+
+{:.good}
+<?code-excerpt "usage_good.dart (late-init-list)"?>
+{% prettify dart tag=pre+code %}
+class Point {
+  double x, y;
+  Point.polar(double theta, double radius)
+      : x = cos(theta) * radius,
+        y = sin(theta) * radius;
+}
+{% endprettify %}
+
+{:.bad}
+<?code-excerpt "usage_bad.dart (late-init-list)"?>
+{% prettify dart tag=pre+code %}
+class Point {
+  late double x, y;
+  Point.polar(double theta, double radius) {
+    x = cos(theta) * radius;
+    y = sin(theta) * radius;
+  }
+}
+{% endprettify %}
+
+
+The initialization list gives you access to constructor parameters and lets you
+initialize fields before they can be read. So, if it's possible to use that,
```

Comment:
```suggestion
initialize fields before they can be read. So, if it's possible to use an initialization list,
```

(There was quite a distance between "that" and its antecedent.)

Also... I just noticed that you use _initialization list_ but the language tour uses _initializer list_. I just checked the language spec, and it uses _initializer list_. Could you make that change throughout Effective Dart?

Reply:
Will do.